From 966a977d734296ebe0d0a86299e83773dfe5c40e Mon Sep 17 00:00:00 2001
From: "Wu, Hao" <hao.wu@intel.com>
Date: Thu, 9 Jul 2015 19:02:40 +0800
Subject: [PATCH] xhci-intel-cap: set mux to device mode as default mode for
 OTG case

When used for OTG purpose, set device mode as default path for
the mux. This can avoid some problems, e.g BIOS can not give the
right default mode for the mux in some cases.

Change-Id: I0747bbc0a67f80852a5b006c03cf53d56e9cd306
Tracked-On: https://jira01.devtools.intel.com/browse/GMINL-13932
Signed-off-by: Wu, Hao <hao.wu@intel.com>
---
 drivers/usb/host/xhci-intel-cap.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/usb/host/xhci-intel-cap.c b/drivers/usb/host/xhci-intel-cap.c
index bb5a7aa046d1..94c72caa8dad 100644
--- a/drivers/usb/host/xhci-intel-cap.c
+++ b/drivers/usb/host/xhci-intel-cap.c
@@ -50,6 +50,12 @@ int xhci_intel_vendor_cap_init(struct xhci_hcd *xhci)
 		retval = otg_set_host(hcd->phy->otg, &hcd->self);
 		if (retval)
 			usb_put_phy(hcd->phy);
+		else {
+			/* if register for OTG succeed, then set mux to device
+			 * mode by default for OTG purpose */
+			xhci_dbg(xhci, "set mux to device mode as default\n");
+			xhci_intel_phy_mux_switch(xhci, 1);
+		}
 	} else {
 		hcd->phy = NULL;
 	}
-- 
1.9.1

